subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/fixtures/ToolingApiBackedGradleExecuter.groovy

```diff
@@ -53,13 +53,19 @@ class ToolingApiBackedGradleExecuter extends AbstractGradleExecuter {
         def args = allArgs
         args.remove("--no-daemon")
 
-        usingToolingConnection(workingDir) { connection ->
-            connection.newBuild()
-                .withArguments(args)
-                .addJvmArguments(jvmArgs)
-                .setStandardOutput(new TeeOutputStream(output, System.out))
-                .setStandardError(new TeeOutputStream(error, System.err))
-                .run()
+        def systemPropertiesBeforeInvocation = new HashMap<String, Object>(System.getProperties())
```

Comment:
I wonder if we need to move this into `usingToolingConnection`, given that the current system property cleanup is in there. And we probably only should mess with the system properties when running using the embedded executer.

What we should do for sure is to remove the current system property cleanup: https://github.com/gradle/gradle/blob/20f814d3f0e26eb080a73e0cac19e7ddf602f226/subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/fixtures/ToolingApiBackedGradleExecuter.groovy#L92-L95